src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceTransaction.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Configuration
+{
+    public class PackageSourceTransaction
```

Comment:
perhaps it's because of my background from applications that use SQL databases, but I would expect a class with the name `Transaction` to be related to committing or rolling back multiple changes in a single atomic operation. Which is not what this class is doing or representing.

Unfortunately naming this is a bit hard. Is `PackageSourceChange` better? `PackageSourceModification`?  `PackageSourceUpdate`?

Reply:
Please avoid the heavily used word, "Update".

Reply:
I like Action (`PackageSourceAction`) which doesn't imply Atomicity.

Reply:
Action implies a method/function to me. 
Change/Modification/Update suggests just data to me. 

I second the transaction comment though.

Reply:
since each of the items in the collection will have the same update settings...perhaps we break it out into some sort of a settings object that only has one instance. we still have a naming challenge, but an easier to solve one.
Perhaps:
SavePackageSources(packageSources, sourcePersistanceSettings);
or:
SavePackageSources(packageSources, sourceUpdateSettings);
or 
similar...